sdks/java/core/src/main/java/org/apache/beam/sdk/runners/TransformHierarchy.java

```diff
@@ -483,12 +402,12 @@ private void setOutput(POutput output) {
       checkState(
           this.outputs == null, "Tried to specify more than one output for %s", getFullName());
       checkNotNull(output, "Tried to set the output of %s to null", getFullName());
-      this.outputs = output.expand();
+      this.outputs = PValues.fullyExpand(output.expand());
```

Comment:
I think this line and the similar one below below made it so a `PCollectionView` is not in `outputs`, so topological traversal does not eagerly process `PCollectionView` as a value, and PCollectionViews also do not have a producer (which is correct for the Beam model). This seems to affect only Dataflow and only because it has special logic which treats PCollectionView as a value in its v1beta3 translation.